src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -494,14 +499,25 @@ public Dataset retrieveByName(String name) {
   }
 
   private Dataset retrieveWorker(DatasetSummary summary) {
+    return retrieveWorker(summary, true, true);
+  }
+
+  private Dataset retrieveWorker(
+      DatasetSummary summary, boolean retrieveRelationship, boolean retrieveAsset) {
     Dataset dataset = null;
     try {
       if (summary != null) {
         summary.storage(storageResourceDao.getStorageResourcesByDatasetId(summary.getId()));
         dataset = new Dataset(summary);
         dataset.tables(tableDao.retrieveTables(dataset.getId()));
-        relationshipDao.retrieve(dataset);
-        assetDao.retrieve(dataset);
+        if (retrieveRelationship) {
```

Comment:
I'm wondering why these weren't joins in the first place instead of separate db calls. I also wonder how many other times we're making multiple db calls in a case where a single query could be used.

But I agree that this is probably the simplest fix for this problem right now